service/service_construction.go

```diff
@@ -621,29 +622,26 @@ func (s ConstructionService) CreateOperationDescription(
 		return nil, fmt.Errorf("invalid number of operations")
 	}
 
-	firstCurrency := operations[0].Amount.Currency
-	secondCurrency := operations[1].Amount.Currency
+	currency := operations[0].Amount.Currency
 
-	if firstCurrency == nil || secondCurrency == nil {
+	if currency == nil || operations[1].Amount.Currency == nil {
 		return nil, fmt.Errorf("invalid currency on operation")
 	}
 
-	if types.Hash(firstCurrency) != types.Hash(secondCurrency) {
+	if !utils.Equal(currency, operations[1].Amount.Currency) {
 		return nil, fmt.Errorf("currency info doesn't match between the operations")
 	}
 
-	if types.Hash(firstCurrency) == types.Hash(mapper.AvaxCurrency) {
-		return s.createOperationDescription(mapper.AvaxCurrency, mapper.OpCall), nil
+	if utils.Equal(currency, mapper.AvaxCurrency) {
+		return s.createOperationDescription(currency, mapper.OpCall), nil
 	}
-	_, firstOk := firstCurrency.Metadata[mapper.ContractAddressMetadata].(string)
-	_, secondOk := secondCurrency.Metadata[mapper.ContractAddressMetadata].(string)
 
-	// Not Native Avax, we require contractInfo in metadata
-	if !firstOk || !secondOk {
-		return nil, fmt.Errorf("non-native currency must have contractAddress in metadata")
+	// ERC-20s must have contract address in metadata
+	if _, ok := currency.Metadata[mapper.ContractAddressMetadata].(string); !ok {
```

Comment:
We should check both currencies

Reply:
we verify they're equivalent in line 631 which includes the `Metadata` map:
```golang
	if !utils.Equal(currency, operations[1].Amount.Currency) {
```

Reply:
Ah fair.